go/ct/common/u256_test.go

```diff
@@ -461,3 +462,213 @@ func TestU256_MarshallingRoundTrip(t *testing.T) {
+	}{
+		{U256{}, true},
+		{NewU256(1), true},
+		{NewU256(1, 2), false},
+		{NewU256(1, 2, 3), false},
+		{NewU256(42, 13, 47, 1), false},
+		{NewU256(0xa000000000000000, 0xb000000000000000, 0xc000000000000000, 0xd000000000000000), false},
+		{NewU256(0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff, 0xffffffffffffffff), false},
+	}
+
+	for _, test := range tests {
+		if want, got := test.want, test.value.IsUint64(); want != got {
+			t.Errorf("Unexpected result from IsUint64, want %v, got %v", want, got)
+		}
+	}
+}
+
+func TestU256_Uint64(t *testing.T) {
+	tests := []struct {
+		value U256
+		want  uint64
+	}{
+		{U256{}, 0},
+		{NewU256(1), 1},
+		{NewU256(0xffffffffffffffff), 0xffffffffffffffff},
+	}
+
+	for _, test := range tests {
+		if want, got := test.want, test.value.Uint64(); want != got {
+			t.Errorf("Unexpected result from Uint64, want %v, got %v", want, got)
+		}
+	}
+}
+
+func TestU256_Uint256Conversion(t *testing.T) {
+	tests := []struct {
+		original  *uint256.Int
+		converted U256
+	}{
+		{uint256.NewInt(0), U256{}},
+		{uint256.NewInt(1), NewU256(1)},
+		{uint256.NewInt(256), NewU256(256)},
+		{new(uint256.Int).Lsh(uint256.NewInt(1), 64), NewU256(1, 0)},
+	}
+
+	for _, test := range tests {
+		if want, got := test.original, test.converted.Uint256(); !want.Eq(&got) {
+			t.Errorf("Unexpected result from Uint256, want %v, got %v", want, got)
+		}
+		if want, got := test.converted, NewU256FromUint256(test.original); !want.Eq(got) {
+			t.Errorf("Unexpected result from NewU256FromUint256, want %v, got %v", want, got)
+		}
+	}
+}
+
+func TestU256_SignExtend(t *testing.T) {
+
+	maxu64 := uint64(0xffffffffffffffff)
+
+	tests := []struct {
+		value    U256
+		argument U256
+		want     U256
+	}{
+		{U256{}, NewU256(1), U256{}},
+		{U256{}, NewU256(0xfff), U256{}},
+		{NewU256(1), NewU256(1), NewU256(1)},
+		{NewU256(1), NewU256(0xfff), NewU256(1)},
+		{NewU256(0xff), NewU256(1), NewU256(0xff)},
+		{NewU256(0xff), NewU256(0xfff), NewU256(0xff)},
+		{NewU256(maxu64), NewU256(1), NewU256(maxu64, maxu64, maxu64, maxu64)},
+		{NewU256(maxu64), NewU256(0xfff), NewU256(maxu64)},
+		{NewU256(maxu64, maxu64, maxu64, maxu64), NewU256(1), NewU256(maxu64, maxu64, maxu64, maxu64)},
+		{NewU256(maxu64, maxu64, maxu64, maxu64), NewU256(0xfff), NewU256(maxu64, maxu64, maxu64, maxu64)},
+	}
+
+	for _, test := range tests {
+		t.Run(test.value.String(), func(t *testing.T) {
+			if want, got := test.want, test.value.SignExtend(test.argument); !want.Eq(got) {
+				t.Errorf("Unexpected result from SignExtend, want %v, got %v", want, got)
+			}
+		})
+	}
+}
+
+func TestU256_And(t *testing.T) {
+	maxu64 := uint64(0xffffffffffffffff)
+	tests := []struct {
+		a    U256
+		b    U256
+		want U256
+	}{
+		{U256{}, U256{}, U256{}},
+		{NewU256(1), NewU256(1), NewU256(1)},
+		{NewU256(0xff), NewU256(0xff), NewU256(0xff)},
+		{NewU256(maxu64), NewU256(maxu64), NewU256(maxu64)},
+		{NewU256(0x12345678), NewU256(0x87654321), NewU256(0x02244220)},
```

Comment:
nit: make the intend explicit
```suggestion
		{NewU256(0x12345678), NewU256(0x87654321), NewU256(0x12345678 & 0x87654321)},
```